src/core/lib/gpr/arena.cc

```diff
@@ -120,51 +120,56 @@ size_t gpr_arena_destroy(gpr_arena* arena) {
 
 void* gpr_arena_alloc(gpr_arena* arena, size_t size) {
   size = GPR_ROUND_UP_TO_ALIGNMENT_SIZE(size);
-  size_t start = static_cast<size_t>(
+  size_t previous_size_of_arena_allocations = static_cast<size_t>(
       gpr_atm_no_barrier_fetch_add(&arena->size_so_far, size));
-  size_t total_size_of_arena_allocations = start + size;
+  size_t updated_size_of_arena_allocations =
+      previous_size_of_arena_allocations + size;
   zone* z = &arena->initial_zone;
-
-  // Check to see if the allocation isn't able to start in the initial zone.
+  // Check to see if the allocation isn't able to end in the initial zone.
   // This statement is true only in the uncommon case because of our arena
   // sizing historesis (that is, most calls should have a large enough initial
   // zone and will not need to grow the arena).
-  if (start >= z->size_end) {
-    // Find the correct zone to start the allocation in
+  if (updated_size_of_arena_allocations > z->size_end) {
+    // Find a zone to fit this allocation
     gpr_mu_lock(&arena->arena_growth_mutex);
-    while (start >= z->size_end) {
+    while (updated_size_of_arena_allocations > z->size_end) {
       if (z->next == nullptr) {
-        size_t next_z_size = total_size_of_arena_allocations;
+        size_t next_z_size = updated_size_of_arena_allocations;
         z->next = static_cast<zone*>(zalloc_aligned(
             GPR_ROUND_UP_TO_ALIGNMENT_SIZE(sizeof(zone)) + next_z_size));
         z->next->size_begin = z->size_end;
         z->next->size_end = z->size_end + next_z_size;
+        // Note that we do an extra increment of size_so_far to prevent multiple
```

Comment:
Suggest moving this new code up to line 137, so that we reset the value of `updated_size_of_arena_allocations` before we use it to set the size of the new zone.

Reply:
Done.

Reply:
Doesn't look like this was actually done.

Reply:
Huh I thought I pushed. I must have made some kind of mistake. It's on my workstation, so I'll take a look tomorrow.

Reply:
Ok I think I fixed it. Should be there now. Thanks for the notification!